### PR TITLE
Time of transactions

### PR DESCRIPTION
**Pull request by MartinOndejka**
Adds time to the diff on da layer. Needed for analytics and metadata for explorer.
Upgrades the da node api to v2.

Order of upgrades:
1. da nodes
2. sequencer
3. archive relay

### CODE REVIEW

**Comment by L-as** [comment]
Why does the DA need to know this? This smells wrong.

**Comment by MartinOndejka** [comment]
> Why does the DA need to know this? This smells wrong.

It's our only persistent storage.

**Comment by L-as** [comment]
I don't see how that means this is the correct solution.

What do you want to use the time information for?

File: src/app/zeko/sequencer/lib/zeko_sequencer.ml [hunk 1]
```diff
@@ -1055,6 +1057,8 @@ let%test_module "Sequencer tests" =
             match%bind
               Da_layer.Client.distribute_genesis_diff ~logger ~config:da_config
                 ~ledger:ephemeral_ledger
+                ~timestamp:
+                  (Block_time.now (Block_time.Controller.basic ~logger))
```
Comment (L-as) [inline src/app/zeko/sequencer/lib/zeko_sequencer.ml]: move to da node as talked about

**Review by L-as — CHANGES_REQUESTED** [review]
DA node should not trust sequencer wrt. timestamp

**Comment by L-as** [comment]
Please don't merge without re-review.

**Pull request merged**